core/src/main/java/org/apache/iceberg/MetadataTable.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg;
+
+/** Interface representing a metadata table. */
+public interface MetadataTable {
```

Comment:
This may not be enough. We may need to check `spec.schema()`. But the overall idea seems still valid.

Reply:
Yea I need the table because Partitioning.partitionType(table) takes a table.  I could add a Partitioning.partitionType(specs).  That being said, it seems a bit misleading to me that spec() of PositionDeletesTable returns spec of underlying table, because the position delete table itself is partitioned a different way than the underlying table, not sure if it's a valid concern.

Reply:
Still had the doubt, but made the change.

Reply:
Well, my original idea was that the position deletes table is actually partitioned in the same way as the main table as data and delete files share the same spec IDs. Let me think for a bit if that's really the case.

Reply:
It is kind of questionable, I guess. On one hand, the position deletes table schema has data columns and delete files use correct spec IDs. On the other hand, it is still a bit different compared to base tables. Let me think.

Reply:
After thinking more about it, would it be a good idea to expose transformed specs? I mean those that we use during planning by calling `BaseMetadataTable.transformSpec`?